src/execution.jl

```diff
@@ -367,6 +367,23 @@ macro belapsed(args...)
     end)
 end
 
+"""
+    @ballocated expression [other parameters...]
+
+Similar to the `@allocated` macro included with Julia,
+this returns the number of bytes allocated when executing
+a given expression.   It uses the `@benchmark`
+macro, however, and accepts all of the same additional
+parameters as `@benchmark`.  The returned allocations
+is for the trial with the *minimum* elapsed time measured
+during the benchmark.
+"""
+macro ballocated(args...)
+    return esc(quote
+        $BenchmarkTools.memory($BenchmarkTools.minimum($BenchmarkTools.@benchmark $(args...)))
```

Comment:
Hmm, I wonder whether minimum should be configurable instead?

Reply:
Maybe maybe not but either way
out of scope for this PR.

That would need to change brime and belapsed also And I would think it's more interesting for them

Reply:
Ah if that's what `@btime` does too then that's fine.